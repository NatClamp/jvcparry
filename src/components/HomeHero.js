import React from 'react'
import { Container, Div, Button, Text } from 'atomize'
import { Link } from 'react-router-dom'

import bg from '../images/bg-min.jpg';
import logoInverted from '../images/logo-inverted.png'
import Modal from '../components/SubscribeModal';


const HomeHero = () => {

  return (
    <>
      <Container minW='100%' minH='100vh' d="flex" flexDir="column" justify="center" align="center" p="2rem" bgImg={bg} bgSize="cover"
        bgPos="center">
        <Modal/>
        <Div d="flex" flexDir="row" justify="center" align="center" w={{ xs: 'auto', md: '300px' }} p={{ b: '30px' }}>
          <img src={logoInverted} alt="JVC Parry logo" style={{ maxHeight: '120px' }} />
        </Div>
        <Text tag='h3' textSize='display1' textColor='white'>
          JVC Parry
        </Text>
        <Text tag='h4' textSize='title' textColor='white'>
          Indie TTRPG creator
        </Text>
        <Div className='homeHeroNav' d='flex' flexDir={{ xs: 'column', sm: 'row' }} justify={{ xs: 'center', sm: 'space-around' }} align={{ xs: 'center', sm: 'space-around' }} w={{ xs: '80%', sm: '30%' }} m={{ t: '30px' }}>
          <Link to='/products' style={{ textDecoration: 'none' }}>
            <Button
              textSize="subheader"
              hoverTextColor="black"
              textColor="black"
              hoverBg="gray400"
              bg="white"
              m='1rem'
              w='10em'
              shadow="4"
              hoverShadow="6"
            >Products
                        </Button>
          </Link>
          <Link to='/hire-me' style={{ textDecoration: 'none' }}>
            <Button
              textSize="subheader"
              hoverTextColor="black"
              textColor="black"
              hoverBg="gray400"
              bg="white"
              m='1rem'
              w='10em'
              shadow="4"
              hoverShadow="6"
            >Hire Me
                        </Button>
          </Link>
          <Link to='/blog' style={{ textDecoration: 'none' }}>
            <Button
              textSize="subheader"
              hoverTextColor="black"
              textColor="black"
              hoverBg="gray400"
              bg="white"
              m='1rem'
              w='10em'
              shadow="4"
              hoverShadow="6"
            >Blog
                        </Button>
          </Link>
        </Div>
      </Container>
    </>
  )
}

export default HomeHero
