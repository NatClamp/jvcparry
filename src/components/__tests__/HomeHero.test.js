/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen, cleanup } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import '@testing-library/jest-dom/extend-expect';

import HomeHero from '../HomeHero';

describe('HomeHero component', () => {
  afterEach(cleanup);

  const history = createMemoryHistory();
  it('should render the HomeHero component', () => {
    render(
      <Router history={history}>
        <HomeHero />
      </Router>,
    );
    expect(screen.getByText(/Indie TTRPG creator/i)).toBeInTheDocument();
  });
  it('should match the snapshot', () => {
    const tree = renderer
      .create(
        <Router history={history}>
          <HomeHero />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
