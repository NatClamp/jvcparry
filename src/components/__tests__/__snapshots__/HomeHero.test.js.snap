// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHero component should match the snapshot 1`] = `
<div
  className=""
>
  <div
    className=""
  >
    <img
      alt="JVC Parry logo"
      src={
        Object {
          "process": [Function],
        }
      }
      style={
        Object {
          "maxHeight": "120px",
        }
      }
    />
  </div>
  <h3
    className=""
  >
    JVC Parry
  </h3>
  <h4
    className=""
  >
    Indie TTRPG creator
  </h4>
  <div
    className="homeHeroNav "
  >
    <a
      href="/products"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className=""
      >
        Products
      </button>
    </a>
    <a
      href="/hire-me"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className=""
      >
        Hire Me
      </button>
    </a>
    <a
      href="/blog"
      onClick={[Function]}
      style={
        Object {
          "textDecoration": "none",
        }
      }
    >
      <button
        className=""
      >
        Blog
      </button>
    </a>
  </div>
</div>
`;
